### PR TITLE
Fix CVE-2007-4559 (tar.extractall)

### PR DESCRIPTION
**Pull request by romain-intel**
See #1177 for more details

### CODE REVIEW

**Review by shrinandj — DISMISSED** [review]

File: metaflow/client/core.py [hunk 1]
```diff
@@ -759,7 +759,8 @@ class MetaflowCode(object):
 
     You can extract code in the directory `snapshot` like so:
     ```
-    Run(...).code.tarball.extractall(path='snapshot')
+    from metaflow.util import tar_safe_extract
```
Comment (savingoyal) [inline metaflow/client/core.py:762]: is this needed? we don't expose any util methods from metaflow.util to the end user. instead we can recommend any method that the user has at their disposal to do tar extraction.

**Review by savingoyal — CHANGES_REQUESTED** [review]

File: metaflow/client/core.py (hunk 1 above)
Comment (romain-intel) [reply metaflow/client/core.py:762]: Sure. I can remove the entire comment and just say that the tarball is given by Run(...).code.tarball and just leave it at that. I didn't want to recommend something that is potentially vulnerable (I know the risk is very very small).

Comment (savingoyal) [reply metaflow/client/core.py:762]: That works!

Comment (savingoyal) [reply metaflow/client/core.py:762]: Please merge the PR after that change is in :)

**Review by savingoyal — APPROVED** [review]

**Pull request merged**